pkg/k8s/cluster.go

```diff
@@ -78,13 +82,34 @@ func NewClusterRequirements() *ClusterRequirements {
 }
 
 func (clusterRequirements ClusterRequirements) Validate(ctx context.Context, client *Client, namespace string) []error {
+	if err := clusterRequirements.validateServerVersion(client); err != nil {
+		return []error{err}
+	}
+
 	if authErrors := clusterRequirements.validateAuthorization(ctx, client, namespace); len(authErrors) > 0 {
 		return authErrors
 	}
 
 	return []error{}
 }
 
+func (clusterRequirements ClusterRequirements) validateServerVersion(client *Client) error {
+	var err error
+
+	var versionInfo *version.Info
+	if versionInfo, err = client.Discovery().ServerVersion(); err != nil {
+		return err
+	}
+
+	serverVersion := semver.MustParse(fmt.Sprintf("%s.%s.0", versionInfo.Major, versionInfo.Minor))
```

Comment:
don't use `MustParse` on runtime vars, on fail it will panic. only use it on predefined strings, so it will panic on dev mode too, if there are any other mustParse on the project which are not used on const change them to `Parse`